Antidote/CallViewController.m

```diff
@@ -0,0 +1,184 @@
+
+@property (strong, nonatomic) OCTSubmanagerCalls *manager;
+@property (strong, nonatomic, readwrite) OCTChat *chat;
+@property (strong, nonatomic) OCTCall *call;
+@property (strong, nonatomic) RBQFetchedResultsController *callController;
+
+@property (strong, nonatomic) UILabel *nameLabel;
+@property (strong, nonatomic) UIButton *cancelCallButton;
+@property (strong, nonatomic) UILabel *timerLabel;
+
+@end
+
+@implementation CallViewController
+
+#pragma mark - Life cycle
+
+- (instancetype)initWithChat:(OCTChat *)chat submanagerCalls:(OCTSubmanagerCalls *)manager
+{
+    self = [super init];
+    if (! self) {
+        return nil;
+    }
+
+    _chat = chat;
+
+    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"chat == %@", chat];
+    _callController = [Helper createFetchedResultsControllerForType:OCTFetchRequestTypeCall predicate:predicate delegate:self];
+    _callController.delegate = self;
+
+    _call = [_callController fetchedObjects].firstObject;
+    if (! _call) {
+        return nil;
+    }
+
+    _manager = manager;
+
+    return self;
+}
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+
+    self.view.opaque = YES;
+
+    [self setupBlurredView];
+    [self createNameLabel];
+
+    [self createEndCallButton];
+    [self createCallTimer];
+
+    [self layoutSubviews];
+}
+
+#pragma mark -  RBQFetchedResultsControllerDelegate
+
+- (void)controllerWillChangeContent:(RBQFetchedResultsController *)controller
+{}
+
+- (void)controller:(RBQFetchedResultsController *)controller didChangeObject:(RBQSafeRealmObject *)anObject atIndexPath:(NSIndexPath *)indexPath forChangeType:(NSFetchedResultsChangeType)type newIndexPath:(NSIndexPath *)newIndexPath
+{}
+
+- (void)controller:(RBQFetchedResultsController *)controller didChangeSection:(RBQFetchedResultsSectionInfo *)section atIndex:(NSUInteger)sectionIndex forChangeType:(NSFetchedResultsChangeType)type
+{}
+
+- (void)controllerDidChangeContent:(RBQFetchedResultsController *)controller
+{
+    self.call = [self.callController fetchedObjects].firstObject;
+
+    self.timerLabel.text = [self convertToTimeFromInterval:self.call.callDuration];
+    [self.timerLabel setNeedsDisplay];
+}
+#pragma mark - Private
+
+- (void)setupBlurredView
+{
+    self.view.backgroundColor = [UIColor clearColor];
+
+    UIView *darkView = [[UIView alloc] initWithFrame:self.view.bounds];
+    darkView.backgroundColor = [UIColor blackColor];
+    darkView.alpha = 0.5;
+
+    [self.view addSubview:darkView];
+
+    UIVisualEffect *blurEffect = [UIBlurEffect effectWithStyle:UIBlurEffectStyleLight];
+
+    UIVisualEffectView *visualEffectView = [[UIVisualEffectView alloc] initWithEffect:blurEffect];
+    visualEffectView.frame = self.view.bounds;
+
+    [self.view addSubview:visualEffectView];
+}
+
+- (void)createNameLabel
+{
+    if (self.call.chat.friends.count == 1) {
+        OCTFriend *friend = self.call.chat.friends.firstObject;
+
+        CGRect labelFrame = CGRectMake(0, 0, 200, 200);
```

Comment:
Please use constraints for that.

Generally speaking, almost all Antidote code **doesn't** use auto layout and sets frames manually. However auto layout is better. That's why we want to use constraints **everywhere** in a new code (and migrate old code after time) :-)